orangecontrib/prototypes/widgets/owsplit.py

```diff
@@ -0,0 +1,135 @@
+import Orange.data
+import numpy as np
+from Orange.data import Table
+from Orange.data.util import SharedComputeValue
+from Orange.widgets import gui
+from Orange.widgets.settings import ContextSetting, DomainContextHandler
+from Orange.widgets.widget import OWWidget, Msg, Output, Input
+from AnyQt.QtCore import Qt
+
+from Orange.data import DiscreteVariable, StringVariable
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.utils.widgetpreview import WidgetPreview
+
+foo = ["foo,bar", "foo,baz", "foo", "bar,baz", "baz,foo", "baz", "bar"]
+
+
+def values_from_string(var_list, splitter):
+    values = [s.split(splitter) for s in var_list]
+    new_values = sorted(set(val for vals in values for val in vals))
+    mapper = {v: i for i, v in enumerate(new_values)}
+
+    X = np.zeros((len(var_list), len(new_values)))
+    for i, vals in enumerate(values):
+        X[i, list(map(mapper.get, vals))] = 1
+    return X, new_values
+
+
+def values_to_string(var_list, values):
+    # turn discrete to string variable
+    return [values[int(i)] if not np.isnan(i) else "nan" for i in var_list]
+
+
+class TransformDomain:
+    # new name so that it doesn't overlap
+    def __init__(self, new_data):
+        self.new_data = new_data
+
+    def __call__(self, data):
+        # make some transformation here
+        if data.domain != self.new_data.domain:
+            data = data.transform(self.new_data.domain)
+        return self.new_data.transform(data.X)
+
+
+class OneHotStrings(SharedComputeValue):
+
+    def __init__(self, fn, feature):
+        super().__init__(fn)
+        self.feature = feature
+
+    def compute(self, _, transformed):
+        return transformed[:, self.feature]
+
+
+class OWSplit(OWWidget):
+    name = "Split"
+    description = "Split string variables to create discrete."
+    icon = "icons/Split.svg"
+    priority = 700
+
+    class Inputs:
+        data = Input("Data", Table)
+
+    class Outputs:
+        data = Output("Data", Table)
+
+    class Warning(OWWidget.Warning):
+        no_disc = Msg("Data contains only numeric variables.")
+
+    want_main_area = False
+    resizing_enabled = False
+
+    settingsHandler = DomainContextHandler()
+    attribute = ContextSetting(None)
+    splitter = ContextSetting(";")
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+
+        gui.lineEdit(
+            self.controlArea, self, "splitter",
+            orientation=Qt.Horizontal, box="Split by", callback=self.apply)
```

Comment:
Add auto comit, use deferred commit, and here set `callback` to `self.apply.deferred`.